src/fidesops/service/privacy_request/request_runner_service.py

```diff
@@ -0,0 +1,147 @@
+import logging
+from datetime import datetime
+from typing import Set
+
+from sqlalchemy.orm import Session
+
+from fidesops import common_exceptions
+from fidesops.graph.graph import DatasetGraph
+from fidesops.models.connectionconfig import ConnectionConfig
+from fidesops.models.datasetconfig import DatasetConfig
+from fidesops.models.policy import ActionType
+from fidesops.models.privacy_request import PrivacyRequest
+from fidesops.service.storage.storage_uploader_service import upload
+from fidesops.task.graph_task import (
+    run_access_request,
+    filter_data_categories,
+    run_erasure,
+)
+from fidesops.util.cache import FidesopsRedis
+
+logging.basicConfig(level=logging.INFO)
+
+
+class PrivacyRequestRunner:
+    """The class responsible for dispatching PrivacyRequests into the execution layer"""
+
+    def __init__(
+        self,
+        cache: FidesopsRedis,
+        db: Session,
+        privacy_request: PrivacyRequest,
+    ):
+        self.cache = cache
+        self.db = db
+        self.privacy_request = privacy_request
+
+    def run(self) -> None:
+        # pylint: disable=too-many-locals
+        """
+        Dispatch a privacy_request into the execution layer by:
+            1. Generate a graph from all the currently configured datasets
+            2. Take the provided identity data
+            3. Start the access request / erasure request execution
+            4. When finished, upload the results to the configured storage destination if applicable
+        """
+
+        logging.info(f"Dispatching privacy request {self.privacy_request.id}")
+        self.start_processing(db=self.db)
+
+        datasets = DatasetConfig.all(db=self.db)
+        dataset_graphs = [dataset_config.get_graph() for dataset_config in datasets]
+        dataset_graph = DatasetGraph(*dataset_graphs)
+        identity_data = self.privacy_request.get_cached_identity_data()
+        connection_configs = ConnectionConfig.all(db=self.db)
+        policy = self.privacy_request.policy
+
+        try:
+            policy.rules[0]
+        except IndexError:
+            raise common_exceptions.MisconfiguredPolicyException(
+                f"Policy with key {policy.key} must contain at least one Rule."
+            )
+
+        try:
+            access_result = run_access_request(
+                privacy_request=self.privacy_request,
+                policy=policy,
+                graph=dataset_graph,
+                connection_configs=connection_configs,
+                identity=identity_data,
+            )
+            if not access_result:
+                logging.info(
+                    f"No results returned for access request {self.privacy_request.id}"
+                )
```

Comment:
Should probably return here if there are no results?

Reply:
we still want to save the processed at time when there are no results for the specific request: https://github.com/ethyca/fidesops/pull/55/files#diff-f0ba0355b20a63eaf049b428f1a3dbddac8567fac5df5f27d0c6a1294066d485R123

Reply:
I think the current functionality is useful here in that it lets the user confirm there are no results -